forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java

```diff
@@ -1,116 +1,185 @@
+				handshakeChannel = (SimpleChannel)handshake;
+				logger.info(RESETMARKER, "Registering forge reset packet.");
+				handshakeChannel.messageBuilder(S2CReset.class, 98)
+						.loginIndex(S2CReset::getLoginIndex, S2CReset::setLoginIndex)
+						.decoder(S2CReset::decode)
+						.encoder(S2CReset::encode)
+						.consumerNetworkThread(HandshakeHandler.biConsumerFor(ClientReset::handleReset))
+						.add();
+				logger.info(RESETMARKER, "Registered forge reset packet successfully.");
+			}
+		}
+		catch (Exception e) {
+			logger.error(RESETMARKER, "Caught exception when attempting to utilize FML's handshake. Disabling mod. Exception: " + e.getMessage());
+		}
+	}
+
+	public static void handleReset(HandshakeHandler handler, S2CReset msg, Supplier<NetworkEvent.Context> contextSupplier) {
+		NetworkEvent.Context context = contextSupplier.get();
+		Connection connection = context.getNetworkManager();
+
+		if (context.getDirection() != NetworkDirection.LOGIN_TO_CLIENT && context.getDirection() != NetworkDirection.PLAY_TO_CLIENT) {
+			connection.disconnect(Component.literal("Illegal packet received, terminating connection"));
+			throw new IllegalStateException("Invalid packet received, aborting connection");
+		}
+
+		logger.info(RESETMARKER, "Received reset packet from server.");
+
+		if (!handleClear(context)) {
+			return;
+		}
+
+		NetworkHooks.registerClientLoginChannel(connection);
+		connection.setProtocol(ConnectionProtocol.LOGIN);
+		connection.setListener(new ClientHandshakePacketListenerImpl(
+				connection, Minecraft.getInstance(), null, statusMessage -> {}
+		));
+
+		context.setPacketHandled(true);
+		try {
+			handshakeChannel.reply(
+				new HandshakeMessages.C2SAcknowledge(),
+				(NetworkEvent.Context)contextConstructor.newInstance(connection, NetworkDirection.LOGIN_TO_CLIENT, 98)
+			);
+		}
+		catch (Exception e) {
+			logger.error(RESETMARKER, "Exception occurred when attempting to reply to reset packet.  Exception: " + e.getMessage());
+			context.setPacketHandled(false);
+			return;
+		}
+		logger.info(RESETMARKER, "Reset complete.");
+	}
+
+	@OnlyIn(Dist.CLIENT)
+	public static boolean handleClear(NetworkEvent.Context context) {
+		CompletableFuture<Void> future = context.enqueueWork(() -> {
+			logger.debug(RESETMARKER, "Clearing");
+
+			// Preserve
+			ServerData serverData = Minecraft.getInstance().getCurrentServer();
+			Pack serverPack = Minecraft.getInstance().getClientPackSource().serverPack;
+
+			// Clear
+			if (Minecraft.getInstance().level == null) {
+				// Ensure the GameData is reverted in case the client is reset during the handshake.
+				GameData.revertToFrozen();
+			}
+			Minecraft.getInstance().getClientPackSource().serverPack = null;
+
+			// Clear
+			Minecraft.getInstance().clearLevel(new GenericDirtMessageScreen(Component.literal("Negotiating..."/*"connect.negotiating"*/)));
+
+			// Restore
+			Minecraft.getInstance().getClientPackSource().serverPack = serverPack;
```

Comment:
Is this actually doing anything? I believe that the variable only gets set in one method which then calls to reload the resource pack so setting this variable without reloading the resource pack either won't do anything, or potentially cause issues?